pints/_nested/_rejection.py

```diff
@@ -34,218 +32,29 @@ class NestedRejectionSampler(pints.NestedSampler):
-        X[0] = 1
-
-        # log marginal likelihood holder
-        v_log_Z = np.zeros(self._iterations + 1)
-
-        # Run
-        i_message = self._active_points - 1
-        for i in range(0, self._iterations):
-            a_running_log_likelihood = np.min(m_active[:, d])
-            a_min_index = np.argmin(m_active[:, d])
-            X[i + 1] = np.exp(-(i + 1) / self._active_points)
-            w[i] = X[i] - X[i + 1]
-            v_log_Z[i] = a_running_log_likelihood
-            m_inactive[i, :] = m_active[a_min_index, :]
-
-            # Independently samples params from the prior until
-            # log_likelihood(params) > threshold.
-            # Note a_running_log_likelihood can be -inf, so while is never run
-            proposed = self._log_prior.sample()[0]
-            log_likelihood = self._log_likelihood(proposed)
-            self._n_evals += 1
-            while log_likelihood < a_running_log_likelihood:
-                proposed = self._log_prior.sample()[0]
-                log_likelihood = self._log_likelihood(proposed)
-                self._n_evals += 1
-            m_active[a_min_index, :] = np.concatenate(
-                (proposed, np.array([log_likelihood])))
-
-            # Show progress
-            if logging:
-                i_message += 1
-                if i_message >= next_message:
-                    # Log state
-                    logger.log(i_message, self._n_evals, timer.time())
-
-                    # Choose next logging point
-                    if i_message > message_warm_up:
-                        next_message = message_interval * (
-                            1 + i_message // message_interval)
-
-        v_log_Z[self._iterations] = logsumexp(m_active[:, d])
-        w[self._iterations:] = float(X[self._iterations]) / float(
-            self._active_points)
-        m_samples_all = np.vstack((m_inactive, m_active))
-        log_Z = logsumexp(v_log_Z, b=w[0:(self._iterations + 1)])
-
-        vP = np.exp(m_samples_all[:, d] - log_Z) * w
-        m_theta = m_samples_all[:, :-1]
-        vIndex = np.random.choice(
-            range(0, self._iterations + self._active_points),
-            self._posterior_samples, p=vP)
-        m_posterior_samples = m_theta[vIndex, :]
-
-        return m_posterior_samples, log_Z
-
-    def set_active_points_rate(self, active_points):
-        """
-        Sets the number of active points for the next run.
-        """
-        active_points = int(active_points)
-        if active_points <= 5:
-            raise ValueError('Number of active points must be greater than 5.')
-        self._active_points = active_points
+        self._proposed = self._log_prior.sample()[0]
+        return self._proposed
 
     def n_hyper_parameters(self):
-        """
-        Returns the number of hyper-parameters for this method (see
-        :class:`TunableMethod`).
-        """
+        """ See :meth:`TunableMethod.n_hyper_parameters()`. """
         return 1
```

Comment:
Is this the same for all nested samplers, or should we be doing this on a per-sampler basis?